docs/plugins/delete-by-query.asciidoc

```diff
@@ -0,0 +1,210 @@
+[[plugins-delete-by-query]]
+== Delete By Query Plugin
+
+The delete by query plugin allows to delete documents from one or more
+indices and one or more types based on a query. It internally uses the
+<<scroll-scan, Scan/Scroll>> and <<docs-bulk, Bulk>> APIs to delete
+documents in an efficient manner.
+
+NOTE: This plugin comes as a replacement of the delete by query API
+removed in Elasticsearch 2.0.
+
+=== Installation
+
+This plugin can be installed using the plugin manager:
+
+[source,sh]
+----------------------------------------------------------------
+$ bin/plugin -i elasticsearch/elasticsearch-delete-by-query
+----------------------------------------------------------------
+
+Note that the node must be restarted to load the plugin. It must
+also be installed on every node of the cluster.
+
+=== Removal
+
+The plugin can be removed with the following command:
+
+[source,sh]
+----------------------------------------------------------------
+$ bin/plugin -r elasticsearch/elasticsearch-delete-by-query
+----------------------------------------------------------------
+
+Note that the node must be stopped before removing the plugin.
+
+=== Usage
+
+The query can either be provided using a simple query string as
+a parameter, or using the <<query-dsl,Query DSL>> defined within
+the request body. Here is an example:
+
+[source,js]
+--------------------------------------------------
+$ curl -XDELETE 'http://localhost:9200/twitter/tweet/_query?q=user:kimchy'
+
+$ curl -XDELETE 'http://localhost:9200/twitter/tweet/_query' -d '{
+    "query" : {
+        "term" : { "user" : "kimchy" }
+    }
+}
+'
+--------------------------------------------------
+
+NOTE: The query being sent in the body must be nested in a `query` key, same as
+the <<search-search,search api>> works
+
+Both above examples end up doing the same thing, which is delete all
+tweets from the twitter index for a certain user.
+
+The result of the commands is:
+
+[source,js]
+--------------------------------------------------
+{
+  "took" : 639,
+  "timed_out" : false,
+  "_indices" : {
+    "_all" : {
+      "found" : 5901,
+      "deleted" : 5901,
+      "missing" : 0,
+      "failed" : 0
+    },
+    "twitter" : {
+      "found" : 5901,
+      "deleted" : 5901,
+      "missing" : 0,
+      "failed" : 0
+    }
+  },
+  "failures" : [ ]
+}
+--------------------------------------------------
+
+Internally, the query is used to execute an initial <<scan,scan>> request. The
+result of this scan request constitutes the set of documents that will be deleted.
+In order to delete this potentially large set of documents, the plugin executes
+multiple <<scroll-scan, scroll>> requests. Each set of scrolled documents will be
+ deleted using a <<docs-bulk, bulk>> request.
+
+Since documents can be updated or deleted by external operations during the
+scan-scroll-bulk process, the plugin keeps track of different counters:
+
+[[delete-by-query-found]]
```

Comment:
I would prefix with `plugins-` to be consistent with the very first tag.
Same for others.